packages/react-core/src/components/DataList/__tests__/DataList.test.tsx

```diff
@@ -12,338 +12,385 @@ import { DataListItemCells } from '../DataListItemCells';
-    expect(screen.getByRole('button')).toHaveAttribute('aria-label', 'Toggle details for');
-    expect(screen.getByRole('button')).toHaveAttribute('aria-expanded', 'false');
-  });
+test(`Cell has ${styles.modifiers.alignRight} when alignRight is passed`, () => {
+  render(
+    <DataListCell data-testid="test" key={0} alignRight>
+      Primary Id
+    </DataListCell>
+  );
+  expect(screen.getByTestId('test')).toHaveClass(styles.modifiers.alignRight);
+});
 
-  test('Toggle expanded', () => {
-    render(<DataListToggle aria-label="Toggle details for" id="ex-toggle2" isExpanded />);
-    expect(screen.getByRole('button')).toHaveAttribute('aria-expanded', 'true');
-  });
+test(`Cell has ${styles.modifiers.icon} when isIcon is passed`, () => {
+  render(
+    <DataListCell data-testid="test" key={0} isIcon>
+      Primary Id
+    </DataListCell>
+  );
+  expect(screen.getByTestId('test')).toHaveClass(styles.modifiers.icon);
+});
 
-  test('DataListAction dropdown', () => {
-    const { asFragment } = render(
-      <DataListAction aria-label="Actions" aria-labelledby="ex-action" id="ex-action" isPlainButtonAction>
-        <Dropdown toggle={(toggleRef) => <MenuToggle ref={toggleRef} />}>
-          <DropdownList>
-            <DropdownItem onClick={jest.fn()} key="action-1">
-              action-1
-            </DropdownItem>
-            ,
-            <DropdownItem onClick={jest.fn()} key="action-2">
-              action-2
-            </DropdownItem>
-          </DropdownList>
-        </Dropdown>
-      </DataListAction>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test(`Cell has ${styles.modifiers.noFill} when isFilled = false`, () => {
+  render(
+    <DataListCell data-testid="test" key={0} isFilled={false}>
+      Primary Id
+    </DataListCell>
+  );
+  expect(screen.getByTestId('test')).toHaveClass(styles.modifiers.noFill);
+});
 
-  test('DataListAction button', () => {
-    const { asFragment } = render(
-      <DataListAction aria-label="Actions" aria-labelledby="ex-action" id="ex-action">
-        <Button id="delete-item-1">Delete</Button>
-      </DataListAction>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Toggle has aria label', () => {
+  render(<DataListToggle aria-label="Toggle details for" id="ex-toggle2" />);
 
-  test('DataListAction visibility - show button when lg', () => {
-    render(
-      <DataListAction
-        visibility={{ default: 'hidden', lg: 'visible' }}
-        aria-labelledby="check-action-item2 check-action-action2"
-        id="check-action-action2"
-        aria-label="Actions"
-      >
-        <Button variant="primary">Primary</Button>
-      </DataListAction>
-    );
+  expect(screen.getByRole('button')).not.toHaveAttribute('aria-labelledby');
+  expect(screen.getByRole('button')).toHaveAttribute('aria-label', 'Toggle details for');
+  expect(screen.getByRole('button')).toHaveAttribute('aria-expanded', 'false');
+});
```

Comment:
We should separate out the other two expectations here and only test for the aria-label rendering. We should also use the `toHaveAccessibleName` rather than toHaveAttribute